File: RoomMate.Domain/Dto/AddressDto.cs
using System.Collections.Generic;

namespace RoomMate.Domain.Dto
{
    public class AddressDto : BaseDto
    {
        public int CityId { get; set; }

        public string CityName { get; set; }

        public string Street { get; set; }

        public string HouseNumber { get; set; }

        public string FlatNumber { get; set; }

        public string FlatName { get; set; }

        public int RoomCount { get; set; }

        public string LoggedUserName { get; set; }

        public ICollection<UserListDto> Users { get; set; }

        public bool IsValid()
        {
            return (IsEmpty(this.Street) && IsEmpty(this.HouseNumber)
                && IsEmpty(this.FlatNumber)
                 && IsValidId(this.CityId));
        }

        public bool DoesFlatExist()
        {
            return (IsEmpty(this.LoggedUserName) && IsValidId(this.Id));
        }
    }
}

File: RoomMate.Front/RoomMatePortal/src/app/address/dto/address-dto.ts
import { UserListDto } from "src/app/user-controll-panel/dto/user-list-dto";
import { RolesEnum } from "src/app/user-controll-panel/dto/RolesEnum";

 export class AddressFlatDto {
  public cityId: number;
  public cityName: string;
  public street: string;
  public houseNumber: string;
  public flatNumber: string;
  public id: number; // flat
  public addressId: number; // addr
  public allAddress: string;
  public flatName: string; // ent
  public roomCount: number; // ent
  public users: UserListDto[];
  public loggedUserName: string; // login
  public roleType: RolesEnum;
  public flatOwnerId: number;
  public area: number; // ent
  public active: boolean;
}
